gateway/health_check_test.go

```diff
@@ -710,6 +710,558 @@ func TestGateway_evaluateHealthChecks(t *testing.T) {
+			},
+			expectedStatus:         http.StatusOK,
+			expectedResponseStatus: Pass,
+		},
+		{
+			name:   "redis health check failed - still returns 200 OK with Fail status",
+			method: http.MethodGet,
+			setupGateway: func(_ *Gateway) {
+			},
+			setupHealthCheck: func(gw *Gateway) {
+				healthInfo := map[string]HealthCheckItem{
+					"redis": {
+						Status:        Fail,
+						ComponentType: Datastore,
+						Output:        "Connection failed",
+					},
+					"dashboard": {
+						Status:        Pass,
+						ComponentType: System,
+					},
+				}
+				gw.setCurrentHealthCheckInfo(healthInfo)
+			},
+			expectedStatus:         http.StatusOK,
+			expectedResponseStatus: Warn,
+		},
+		{
+			name:   "all health checks failed - returns 200 OK with Fail status",
+			method: http.MethodGet,
+			setupGateway: func(_ *Gateway) {
+			},
+			setupHealthCheck: func(gw *Gateway) {
+				healthInfo := map[string]HealthCheckItem{
+					"redis": {
+						Status:        Fail,
+						ComponentType: Datastore,
+						Output:        "Redis connection failed",
+					},
+					"dashboard": {
+						Status:        Fail,
+						ComponentType: System,
+						Output:        "Dashboard connection failed",
+					},
+				}
+				gw.setCurrentHealthCheckInfo(healthInfo)
+			},
+			expectedStatus:         http.StatusOK,
+			expectedResponseStatus: Fail,
+		},
+		{
+			name:   "partial health check failures - returns 200 OK with Warn status",
+			method: http.MethodGet,
+			setupGateway: func(_ *Gateway) {
+			},
+			setupHealthCheck: func(gw *Gateway) {
+				healthInfo := map[string]HealthCheckItem{
+					"redis": {
+						Status:        Pass,
+						ComponentType: Datastore,
+					},
+					"dashboard": {
+						Status:        Fail,
+						ComponentType: System,
+						Output:        "Dashboard connection failed",
+					},
+					"rpc": {
+						Status:        Pass,
+						ComponentType: System,
+					},
+				}
+				gw.setCurrentHealthCheckInfo(healthInfo)
+			},
+			expectedStatus:         http.StatusOK,
+			expectedResponseStatus: Warn,
+		},
+		{
+			name:   "empty health check data - returns 200 OK with Pass status",
+			method: http.MethodGet,
+			setupGateway: func(_ *Gateway) {
+			},
+			setupHealthCheck: func(gw *Gateway) {
+				gw.setCurrentHealthCheckInfo(map[string]HealthCheckItem{})
+			},
+			expectedStatus:         http.StatusOK,
+			expectedResponseStatus: Pass,
+		},
+		{
+			name:   "hide generator header enabled - returns 200 OK",
+			method: http.MethodGet,
+			setupGateway: func(gw *Gateway) {
+				conf := gw.GetConfig()
+				conf.HideGeneratorHeader = true
+				gw.SetConfig(conf)
+			},
+			setupHealthCheck: func(gw *Gateway) {
+				healthInfo := map[string]HealthCheckItem{
+					"redis": {
+						Status:        Pass,
+						ComponentType: Datastore,
+					},
+				}
+				gw.setCurrentHealthCheckInfo(healthInfo)
+			},
+			expectedStatus:         http.StatusOK,
+			expectedResponseStatus: Pass,
+		},
+	}
+
+	for _, tt := range tests {
```

Comment:
please check if `tt := tt` still needs to be done because of goroutine issues that can appear